File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.py
height = []

for _ in range(9) :
    height.append(int(input()))

height.sort()

sum = 0
findDwarf = False

while not findDwarf :
    for a in range(9):
        for b in range(9) :
            if a == b : continue

            sum = 0
            
            for k in range(9) :
                if k in (a,b) : continue
                else : sum += height[k]
            
            if sum == 100 :
                findDwarf = True
                for k in range(9) :
                    if k in (a,b) : continue
                    else : print(height[k])
                break
        if findDwarf : break

File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.cc
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    vector<int> array(9);
    int sum = 0;
    int n;

    for (int i = 0; i < 9; i++) {
        cin >> n;
        array[i] = n;
        sum += n;
    }

    for (int i = 0; i < 8; i++) {
        for (int j = i+1; j < 9; j++) {
            if (sum - array[i] - array[j] == 100) {
                array.erase(array.begin()+i);
                array.erase(array.begin()+j-1);
                
                i += 9;
                break;
            }
        }
    }

    sort(array.begin(), array.end());

    for (int i = 0; i < 7; i++) {
        cout << array[i] << '\n';
    }
}